OPHD/StructureManager.h

```diff
@@ -52,6 +53,79 @@ class StructureManager
 
 	void serialize(NAS2D::Xml::XmlElement* element);
 
+	/**
+	 * Associate a StructureComponent with a structure.
+	 * This transfers ownership of the component instance to the
+	 * structure manager. The component will be destroyed when the
+	 * structure with which it is associated is removed.
+	 */
+	template<typename ComponentTy>
+	void attachComponent(SKey s, ComponentTy* component)
+	{
+		auto& table = mComponents[ComponentTy::componentTypeID];
+		bool success = table.insert(std::make_pair(s, static_cast<StructureComponent*>(component))).second;
+#if defined(_DEBUG)
+		if (!success)
+		{
+			throw std::runtime_error("Structure::Attach() was called on a Structure that already had the component!");
+		}
+#endif
```

Comment:
More `#define`, which I recommend avoiding. I don't think it's necessary here. Just throw the exception for all configurations.

Reply:
sure